DashWallet/Sources/UI/SwiftUI Components/Toast.swift

```diff
@@ -0,0 +1,158 @@
+//
+
+import SwiftUI
+import UIKit
+
+struct ToastView: View {
+    var text: String
+    var icon: IconName? = nil
+    var actionText: String? = nil
+    var action: (() -> Void)? = nil
+    var closeButtonIcon: IconName? = nil
+    var closeAction: (() -> Void)? = nil
+    
+    var body: some View {
+        HStack(spacing: 0) {
+            if let icon = icon {
+                VStack {
+                    Icon(name: icon)
+                        .padding(.leading, 8)
+                        .padding(.top, 12)
+                        .font(.system(size: 15))
+                    Spacer()
+                }
+            }
+            
+            Text(text)
+                .font(.system(size: 14))
+                .lineSpacing(3)
+                .padding(.leading, 8)
+                .padding(.vertical, 12)
+            Spacer()
+            
+            if let text = actionText, let action = action {
+                DashButton(text: text, action: action, style: .plain, size: .extraSmall)
+                    .overrideForegroundColor(Color.background)
+            }
+            
+            if let icon = closeButtonIcon, let action = closeAction {
+                DashButton(leadingIcon: icon, action: action, style: .plain, size: .small)
+                    .overrideForegroundColor(Color.background)
+            }
+        }
+        .padding(.horizontal, 8)
+        .foregroundColor(Color.background)
+        .background(Color.primaryText.opacity(0.9))
+        .cornerRadius(10)
+    }
+}
+
+class ToastHostingView: UIView {
+    private var hostingController: UIHostingController<ToastView>?
+    
+    init(text: String,
+         icon: IconName? = nil,
+         actionText: String? = nil,
+         action: (() -> Void)? = nil,
+         closeButtonIcon: IconName? = nil,
+         closeAction: (() -> Void)? = nil
+    ) {
+        super.init(frame: .zero)
+        
+        let toastView = ToastView(
+            text: text,
+            icon: icon,
+            actionText: actionText,
+            action: action,
+            closeButtonIcon: closeButtonIcon,
+            closeAction: closeAction
+        )
+        hostingController = UIHostingController(rootView: toastView)
+        
+        guard let hostingView = hostingController?.view else { return }
+        hostingView.translatesAutoresizingMaskIntoConstraints = false
+        hostingView.backgroundColor = UIColor.clear
+        addSubview(hostingView)
+        
+        NSLayoutConstraint.activate([
+            hostingView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            hostingView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            hostingView.topAnchor.constraint(equalTo: topAnchor),
+            hostingView.bottomAnchor.constraint(equalTo: bottomAnchor)
+        ])
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+extension UIViewController {
+    func showToast(
```

Comment:
Toast showing logic is very simple now and needs to be invoked from the correct VC. This can be improved in the future so that it can be shown from anywhere.